geozero/src/csv/csv_error.rs

```diff
@@ -0,0 +1,14 @@
+//! CSV Error type.
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum CsvError {
+    #[error("column not found or null")]
+    ColumnNotFound,
+    #[error("Invalid UTF-8 encoding")]
+    InvalidUtf8,
+    #[error("error processing dataset: `{0}`")]
+    Processing(String),
+    #[error("error parsing to WKT `{0}`")]
+    WktError(String),
```

Comment:
#138 wasn't my request, but my understanding is that the point was to move from `String` errors to structured errors, but this is still a String error.

Reply:
I'm using a `String` buffer here in order to display the error message because `Wkt` from the `wkt` crate uses a `&'static str` for errors.

Reply:
Ah, so then making it more structured is out of your hands.  

But there's no value in copying the &'static str to a String is there? In that case, it's preferable to `WktError(&'static str)` since it'll avoid duplicating the contents of the &str onto the heap.